daisy_workflows/export/export_disk_ext.sh

```diff
@@ -14,31 +14,55 @@
 # limitations under the License.
 set -x
 
+BYTES_1GB=1073741824
 URL="http://metadata/computeMetadata/v1/instance/attributes"
 GS_PATH=$(curl -f -H Metadata-Flavor:Google ${URL}/gcs-path)
 FORMAT=$(curl -f -H Metadata-Flavor:Google ${URL}/format)
+DISK_RESIZING_MON=$(curl -f -H Metadata-Flavor:Google ${URL}/resizing-script-name)
 
 # Strip gs://
-LOCAL_PATH=${GS_PATH##*//}
+IMAGE_OUTPUT_PATH=${GS_PATH##*//}
 # Create dir for output
-OUTS_PATH=${LOCAL_PATH%/*}
+OUTS_PATH=${IMAGE_OUTPUT_PATH%/*}
 mkdir -p "/gs/${OUTS_PATH}"
 
-# Disk image size info.
-SIZE_BYTES=$(qemu-img info --output "json" /dev/sdb | grep -m1 "virtual-size" | grep -o '[0-9]\+')
- # Round up to the next GB.
-SIZE_GB=$(awk "BEGIN {print int((${SIZE_BYTES}/1000000000)+ 1)}")
+# Prepare disk size info.
+# 1. Disk image size info.
+SIZE_BYTES=$(lsblk /dev/sdb --output=size -b | sed -n 2p)
+# 2. Round up to the next GB.
+SIZE_OUTPUT_GB=$(awk "BEGIN {print int(((${SIZE_BYTES}-1)/${BYTES_1GB}) + 1)}")
+# 3. Add 5GB of additional space to max size to prevent the corner case that output
+# file is slightly larger than source disk.
+MAX_BUFFER_DISK_SIZE_GB=$(awk "BEGIN {print int(${SIZE_OUTPUT_GB} + 5)}")
 
-echo "GCEExport: Exporting disk of size ${SIZE_GB}GB and format ${FORMAT}."
+# Prepare buffer disk. We have confidence that it won't fail in this step.
+echo "GCEExport: Initializing buffer disk for qemu-img output..."
+sudo mkfs.ext4 /dev/sdc
```

Comment:
what user wll this script be invoked as? (i.e. is `sudo` necessary?)
what error handling is there for these two..what error will user see if they fail?

Reply:
As discussed, to avoid confusion output, added a failure path here.